http-service-hyper/src/lib.rs

```diff
@@ -5,219 +5,59 @@
-        }
-    }
-}
-
-impl<I: TryStream, Sp> Builder<I, Sp> {
-    /// Sets the [`Spawn`] to deal with starting connection tasks.
-    pub fn with_spawner<Sp2>(self, new_spawner: Sp2) -> Builder<I, Sp2> {
-        Builder {
-            inner: self.inner.executor(Compat::new(new_spawner)),
-        }
-    }
-
-    /// Consume this [`Builder`], creating a [`Server`].
-    ///
-    /// # Examples
-    ///
-    /// ```no_run
-    /// use http_service::{Response, Body};
-    /// use http_service_hyper::Server;
-    /// use romio::TcpListener;
-    ///
-    /// // Construct an executor to run our tasks on
-    /// let mut pool = futures::executor::ThreadPool::new()?;
-    ///
-    /// // And an HttpService to handle each connection...
-    /// let service = |req| {
-    ///     futures::future::ok::<_, ()>(Response::new(Body::from("Hello World")))
-    /// };
-    ///
-    /// // Then bind, configure the spawner to our pool, and serve...
-    /// let addr = "127.0.0.1:3000".parse()?;
-    /// let mut listener = TcpListener::bind(&addr)?;
-    /// let server = Server::builder(listener.incoming())
-    ///     .with_spawner(pool.clone())
-    ///     .serve(service);
-    ///
-    /// // Finally, spawn `server` onto our executor...
-    /// pool.run(server)?;
-    /// # Ok::<(), Box<dyn std::error::Error>>(())
-    pub fn serve<S: HttpService>(self, service: S) -> Server<I, S, Sp>
-    where
-        I: TryStream + Unpin,
-        I::Ok: AsyncRead + AsyncWrite + Send + Unpin + 'static,
-        I::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
-        Sp: Clone + Send + 'static,
-        for<'a> &'a Sp: Spawn,
-    {
-        Server {
-            inner: Compat01As03::new(self.inner.serve(WrapHttpService {
-                service: Arc::new(service),
-            })),
+    let s = Arc::new(s);
+
+    let make_svc = make_service_fn({
+        let s = s.clone();
+        move |_| {
+            let s = s.clone();
+            async move {
+                Ok::<_, Box<dyn std::error::Error + Send + Sync>>(service_fn({
+                    move |req: http::Request<hyper::Body>| {
+                        let s = s.clone();
+                        async move {
+                            let mut conn = s.connect().await?;
+                            let (parts, body) = req.into_parts();
+                            let body = http_service::Body::from_stream(
+                                body.map_ok(|b| b.into()).map_err(|e: hyper::Error| {
+                                    std::io::Error::new(std::io::ErrorKind::Other, e)
+                                }),
+                            );
+
+                            let rsp = s
+                                .respond(&mut conn, http::Request::from_parts(parts, body))
+                                .await?;
+
+                            let (parts, body) = rsp.into_parts();
+                            let body: hyper::Body = body.into_vec().await?.into();
```

Comment:
Shouldn't the body be wrapped into a compatible stream for `hyper::Body` instead of converting it into a `Vec`? I see for that we need the `unstable-stream` feature from hyper.